src/bsp/platform/stm32f4_nucleo_wifi/xi_bsp_io_net_stm32f4_nucleo_wifi.c

```diff
@@ -198,28 +219,21 @@ void xi_bsp_io_net_socket_data_received_proxy( uint8_t socket_id,
                                                uint32_t chunk_size )
 {
     ( void )socket_id;
-    if ( xi_wifi_buffer.size == 0 )
-    {
-        if ( chunk_size > XI_BSP_IO_NET_BUFFER_SIZE )
-        {
-            // payload is too big
-            xi_wifi_state = xi_wifi_state_error;
-        }
-        else
-        {
-            memcpy( xi_wifi_buffer.bytes, data_ptr, message_size );
-            xi_wifi_buffer.size = chunk_size;
-        }
-    }
+    ( void )message_size;
+
+    xi_data_desc_t* tail = xi_make_desc_from_buffer_copy( data_ptr , chunk_size );
+
+    if ( xi_net_state.head == NULL ) xi_net_state.head = tail;
```

Comment:
pls use curly brances for all oneliners too. (clang-format would solve this as well)